ietf_gh_utils.py

```diff
@@ -20,23 +20,38 @@ def verify_users(g, user):
             raise SystemExit('User "%s" does not exist"' % (u,))
     return 1
 
-def gh_login(np):
-    """Login to Github  |np| is colon-separated name and password, if omitted
-    either one will be prompted-for.  Returns the handle and the username"""
-    name = None
-    passwd = None
-    if np is not None:
-        name, passwd = np.split(':', 1)
-    if not name:
-        pass
-    if not passwd:
-        pass
-    g = Github(name, passwd)
+def add_gh_auth_arguments(parser):
+    """Add arguments needed for logging into Github to the
+    argparse.ArgumentParser."""
+    group = parser.add_argument_group('Github Auth')
+    group.add_argument('--user', '-u', metavar='GH_ID', required=True,
+                       help="Github username")
+    pw_or_token = group.add_mutually_exclusive_group(required=True)
+    pw_or_token.add_argument('--passwd', '-p',
+                             help="Github password, " +
+                             "if the user doesn't have 2FA enabled")
+    pw_or_token.add_argument('--token', '-t',
+                             help="Github access token with the public_repo scope: " +
+                             "https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line")
+
+def gh_login(args):
+    """Login to Github  |args| is the result of parsing command-line arguments
+    with the parser passed to add_gh_auth_arguments().
+    If both are omitted, username and password will be prompted-for.  Returns
+    the handle and the username"""
+    if args.token is not None:
+        g = Github(args.token)
+    else:
+        if not args.user:
```

Comment:
I suppose I'd prefer a log message, but that can be done separately.  More important to get this in for 2FA support :)\

Reply:
Yep, I just wanted to keep this change as minimal as possible.